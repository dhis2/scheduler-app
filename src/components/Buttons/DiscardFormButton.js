import React, { useState } from 'react'
import { string, bool } from '@dhis2/prop-types'
import { Button } from '@dhis2/ui'
import history from '../../services/history'
import { DiscardFormModal } from '../Modal'

const DiscardFormButton = ({ shouldConfirm, children }) => {
    const [showModal, setShowModal] = useState(false)
    const onClick = shouldConfirm
        ? () => setShowModal(true)
        : () => history.push('/')

    return (
        <React.Fragment>
            <Button onClick={onClick}>{children}</Button>
            {showModal && (
                <DiscardFormModal hideModal={() => setShowModal(false)} />
            )}
        </React.Fragment>
    )
}

DiscardFormButton.defaultProps = {
    shouldConfirm: false,
}

DiscardFormButton.propTypes = {
    children: string.isRequired,
    shouldConfirm: bool,
}

export default DiscardFormButton
