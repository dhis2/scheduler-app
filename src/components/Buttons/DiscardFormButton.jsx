import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { Button } from '@dhis2/ui'
import history from '../../services/history'
import { DiscardFormModal } from '../Modal'

const DiscardFormButton = ({ shouldConfirm, children, small, className }) => {
    const [showModal, setShowModal] = useState(false)
    const onClick = shouldConfirm
        ? () => setShowModal(true)
        : () => history.push('/')

    return (
        <React.Fragment>
            <Button onClick={onClick} small={small} className={className}>
                {children}
            </Button>
            {showModal && (
                <DiscardFormModal
                    hideModal={
                        /* istanbul ignore next */
                        () => setShowModal(false)
                    }
                />
            )}
        </React.Fragment>
    )
}

DiscardFormButton.defaultProps = {
    className: '',
    shouldConfirm: false,
    small: false,
}

const { string, bool } = PropTypes

DiscardFormButton.propTypes = {
    children: string.isRequired,
    className: string,
    shouldConfirm: bool,
    small: bool,
}

export default DiscardFormButton
