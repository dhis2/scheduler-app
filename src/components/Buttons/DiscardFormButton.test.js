import React from 'react'
import { shallow, mount } from 'enzyme'
import history from '../../services/history'
import { DumbDiscardFormButton as DiscardFormButton } from './DiscardFormButton'

describe('<DiscardFormButton>', () => {
    beforeEach(() => {
        history.push = jest.fn()
    })

    it('renders correctly', () => {
        const wrapper = shallow(
            <DiscardFormButton showModal={() => {}} shouldConfirm={false}>
                Discard
            </DiscardFormButton>
        )

        expect(wrapper).toMatchSnapshot()
    })

    it('calls showModal correctly when clicked and it should confirm', () => {
        const showModal = jest.fn()
        const wrapper = mount(
            <DiscardFormButton showModal={showModal} shouldConfirm>
                Discard
            </DiscardFormButton>
        )

        wrapper.find('button').simulate('click')

        expect(showModal.mock.calls[0][0]).toMatchSnapshot()
    })

    it('calls history correctly when clicked and it should not confirm', () => {
        const wrapper = mount(
            <DiscardFormButton showModal={() => {}} shouldConfirm={false}>
                Discard
            </DiscardFormButton>
        )

        wrapper.find('button').simulate('click')

        expect(history.push.mock.calls[0][0]).toMatchSnapshot()
    })
})
