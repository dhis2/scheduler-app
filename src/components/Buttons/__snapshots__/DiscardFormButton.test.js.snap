// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiscardFormButton> calls history correctly when clicked and it should not confirm 1`] = `"/"`;

exports[`<DiscardFormButton> calls showModal correctly when clicked and it should confirm 1`] = `
Object {
  "type": "DISCARD_FORM",
}
`;

exports[`<DiscardFormButton> renders correctly 1`] = `
<Button
  dataTest="dhis2-uicore-button"
  onClick={[Function]}
  type="button"
>
  Discard
</Button>
`;
